react/src/components/Table/ColumnVisibilityModal.tsx

```diff
@@ -1,70 +1,152 @@
 import { useState } from 'react';
+import { DragDropContext, Draggable, Droppable, DropResult } from 'react-beautiful-dnd';
 import { BsFillEyeFill, BsFillEyeSlashFill } from 'react-icons/bs';
-import { ColumnInstance, HeaderGroup, Row, UseTableInstanceProps } from 'react-table';
+import { MdDragHandle } from 'react-icons/md';
+import { ColumnInstance, HeaderGroup, UseTableInstanceProps } from 'react-table';
 import { camelize } from '../../utils';
-import { Button, Checkbox, InlineFlex, Modal } from '../index';
+import { Button, Checkbox, Flex, InlineFlex, Modal } from '../index';
 import { IconPadder } from './Table.styles';
 
 interface ColumnVisibilityModalProps<T extends object>
     extends Pick<UseTableInstanceProps<T>, 'toggleHideColumn'> {
-    rows: Row<T>[];
-    visibleColumns: ColumnInstance<T>[];
     headerGroups: HeaderGroup<T>[];
     toggleGroupVisibility: (g: HeaderGroup<T>) => void;
+    allColumns: ColumnInstance<T>[];
+    setColumnOrder: (update: string[] | ((columnOrder: string[]) => string[])) => void;
 }
 
 export default function ColumnVisibilityModal<T extends {}>({
     headerGroups,
     toggleGroupVisibility,
     toggleHideColumn,
+    allColumns,
+    setColumnOrder,
 }: ColumnVisibilityModalProps<T>) {
     const [showModal, setShowModal] = useState<boolean>(false);
+    const [order, setOrder] = useState<ColumnInstance<T>[]>([]);
+    const reorder = (prevPos: number, newPos: number): ColumnInstance<T>[] => {
+        const result: ColumnInstance<T>[] = order;
+        const [removed] = result.splice(prevPos, 1);
+        result.splice(newPos, 0, removed);
+        return result;
+    };
+    const onDragEnd = (result: DropResult) => {
+        const { source, destination } = result;
+        if (!destination) {
+            return;
+        }
+        const columnOrder = reorder(source.index, destination.index);
+        setOrder(columnOrder);
+    };
 
     return (
         <InlineFlex>
-            <Button variant="secondary" onClick={() => setShowModal(!showModal)}>
+            <Button
+                variant="secondary"
+                onClick={() => {
+                    setShowModal(!showModal);
+                    setOrder(allColumns);
+                }}
+            >
                 Customize columns
                 <IconPadder>{showModal ? <BsFillEyeFill /> : <BsFillEyeSlashFill />}</IconPadder>
             </Button>
             <Modal
                 active={showModal}
                 hideModal={() => setShowModal(false)}
                 title="Customize Columns"
+                footer="Apply"
+                onClick={() => {
+                    setColumnOrder(order.map(o => o.id));
+                    setShowModal(false);
+                }}
+                helperText="Only columns within the same group can be reordered."
```

Comment:
Because the "Customize Columns" module is for both toggling visibility and moving columns around, can we adjust the helperText to reflect that? And we can then add "note that only columns within the same group can be reordered"